20150310-wp-core-contrib-day.md

```diff
@@ -0,0 +1,255 @@
+
+While you're at it, why not copy the `wp-tests-config-sample.php` file to `wp-tests-config.php` and update it with the database and database user details for another database you've also created locally so that you can run the unit tests too? It's important to use a separate database for the unit tests as the database is completely refreshed during a unit tests run.
+
+Now you have a `wp-config.php` for your development site, it's time to set up a website such as http://wordpress-core.dev that uses the `build` directory as its document root. I use [MAMP Pro](http://www.mamp.info/en/mamp-pro/) for its simplicity and flexibility, but you should use whatever you're most comfortable with. Ideally use PHP 5.2.x in your development as this is the minimum supported version of PHP for WordPress so you don't want to use features from later versions of PHP.
+
+![WordPress Core Site Setup (MAMP PRO)](https://s3.amazonaws.com/cdn.deliciousbrains.com/content/uploads/2015/03/WordPress-Core-Site-Setup-MAMP-PRO.png)
+
+### Give It a Spin
+The moment of truth, make sure you can connect to your new WordPress development site and set it up.
+
+![WordPress › Installation](https://s3.amazonaws.com/cdn.deliciousbrains.com/content/uploads/2015/03/WordPress-%E2%80%BA-Installation.png)
+
+![Dashboard ‹ WordPress Core — WordPress](https://s3.amazonaws.com/cdn.deliciousbrains.com/content/uploads/2015/03/Dashboard-%E2%80%B9-WordPress-Core-—-WordPress.png)
+
+![WordPress Core | Just another WordPress site](https://s3.amazonaws.com/cdn.deliciousbrains.com/content/uploads/2015/03/WordPress-Core-Just-another-WordPress-site.png)
+
+### Run The Unit Tests
+It's a very good idea to run the unit tests just to be sure all is well in your development site, and to make sure any changes you later make do not break anything.
+
+From the root of the checked out source...
+
+```
+phpunit
+```
+
+![WordPress Core Unit Tests Run](https://s3.amazonaws.com/cdn.deliciousbrains.com/content/uploads/2015/03/WordPress-Core-Unit-Tests-Run.png)
+
+Don't be surprised if you see a failure or two in the unit tests, during development there often are some, maybe you could fix them? At least you know where the problems exist and so when you make your changes you can tell whether you made things better, worse, or at least didn't affect them!
+
+## Development
+You're now all set up to develop on WordPress Core, keep your repository up to date with `svn up` and rebuild your WordPress Core site with `grunt` as required or through a `watch` process.
+
+Let's assume you're starting off your WordPress Core contributions by refreshing a patch, this entails downloading the patch file from a ticket, applying it to your checked out source, checking that it still applies correctly and does what is expected against the latest development version of WordPress Core, and then re-creating the patch and attaching to the ticket with an updated name and updated keywords.
```

Comment:
Run on sentence. Reworked:

> Let's assume you're starting off your WordPress Core contributions by refreshing a patch. This entails downloading the patch file from a ticket, applying it to your checked out source, checking that it still applies correctly and does what is expected, recreating the patch, and attaching it to the ticket with an updated name and updated keywords.

You might also want to mention about the typical naming scheme: <ticket_num>.<patch_version>.diff